app/src/main/java/foi/air/szokpt/viewmodels/TransactionsViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package foi.air.szokpt.viewmodels
+
+import androidx.compose.ui.graphics.Color
+import androidx.lifecycle.ViewModel
+import foi.air.szokpt.helpers.TransactionsSuccessHandler
+import foi.air.szokpt.ui.theme.TextGray
+import foi.air.szokpt.ui.theme.danger
+import foi.air.szokpt.ui.theme.success
+import hr.foi.air.core.transactions.TransactionsSuccessOutcomeListener
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class TransactionsViewModel : ViewModel() {
+    private val _pieChartData = MutableStateFlow(listOf<PieChartModel>())
+    val pieChartData: StateFlow<List<PieChartModel>> get() = _pieChartData
+
+    private val _errorMessage = MutableStateFlow<String?>(null)
+    val errorMessage: StateFlow<String?> get() = _errorMessage
+
+    private val transactionsHandler = TransactionsSuccessHandler()
+
+    fun fetchTransactions() {
+        _errorMessage.value = null
+
+        transactionsHandler.getTransactionsSuccess(object :
+            TransactionsSuccessOutcomeListener {
+            override fun onSuccessfulTransactionsFetch(
+                totalTransactions: Int,
+                successfulTransactions: Int,
+                rejectedTransactions: Int,
+                canceledTransactions: Int
+            ) {
+                val total = (successfulTransactions + canceledTransactions + rejectedTransactions).toFloat()
```

Comment:
Potencijalno može rezultirati sa 0. U slučaju da se to desi može se desiti nevažeća operacija dijeljenja s 0.